tesseract_kinematics/kdl/include/tesseract_kinematics/kdl/kdl_utils.h

```diff
@@ -102,6 +102,22 @@ struct KDLChainData
   std::vector<std::pair<std::string, std::string>> chains; /**< The chains used to create the object */
 };
 
+/**
+ * @brief The KDLConfig struct
+ *
+ * This contains parameters that can be used to customize the KDL solver for your application.
+ * They are ultimately passed to the constuctor of the undelying ChainIkSolver.
+ *
+ * The defaults provided here are the same defaults imposed by the KDL library.
+ */
+struct KDLConfig
```

Comment:
Does this config only apply to KDL LMA?

Reply:
Yes; the other solvers take similar, but not exactly the same, parameters

Reply:
Michael is correct. I actually started down the path of using this struct for the other KDL factories but stopped when I noticed the different defaults for each.

How would you like me to proceed? Use the same struct but change the defaults to match the constructor for each version? (NR and Fwd)

Reply:
I am fine with creating a different config for each which have different default and moving the struct into the header of the different ones.

Reply:
``` c++
struct KDLInvKinChainLMAConfig
{
}
```

Or you can define the struct in the the KDLInvKinChainLMA class where it would be `struct Config`.

Reply:
@marip8 Do you have a preference?

Reply:
I think having separate configuration structs would be good. I usually embed that struct inside the corresponding class, but either way is fine with me

Reply:
Okay, @seanmc602 lets embed the config struct in the corresponding class.